src/engine/controls/cuecontrol.cpp

```diff
@@ -22,19 +23,39 @@ static const double CUE_MODE_NUMARK = 3.0;
 static const double CUE_MODE_MIXXX_NO_BLINK = 4.0;
 static const double CUE_MODE_CUP = 5.0;
 
+constexpr double kNoColorControlValue = -1;
+
+namespace {
+
+// Helper function to convert doubles into RgbColor instances (or nullopt)
+inline mixxx::RgbColor::optional_t doubleToRgbColor(double value) {
+    if (value < 0) {
+        return std::nullopt;
+    }
+    mixxx::RgbColor::code_t colorCode = static_cast<mixxx::RgbColor::code_t>(value);
```

Comment:
Instead of repeating the longish type twice on a single line `auto` would be an appropriate choice here